cli/azd/cmd/auto_install.go

```diff
@@ -0,0 +1,281 @@
+	console.Message(ctx,
+		fmt.Sprintf("Extension: %s (%s)\n", extension.DisplayName, extension.Description))
+	shouldInstall, err := console.Confirm(ctx, input.ConsoleOptions{
+		DefaultValue: true,
+		Message:      "Would you like to install it?",
+	})
+	if err != nil {
+		return false, nil
+	}
+
+	if !shouldInstall {
+		return false, nil
+	}
+
+	// Install the extension
+	console.Message(ctx, fmt.Sprintf("Installing extension '%s'...\n", extension.Id))
+	filterOptions := &extensions.FilterOptions{}
+	_, err = extensionManager.Install(ctx, extension.Id, filterOptions)
+	if err != nil {
+		return false, fmt.Errorf("failed to install extension: %w", err)
+	}
+
+	console.Message(ctx, fmt.Sprintf("Extension '%s' installed successfully!\n", extension.Id))
+	return true, nil
+}
+
+// ExecuteWithAutoInstall executes the command and handles auto-installation of extensions for unknown commands.
+func ExecuteWithAutoInstall(ctx context.Context, rootContainer *ioc.NestedContainer) error {
+	// Creating the RootCmd takes care of registering common dependencies in rootContainer
+	rootCmd := NewRootCmd(false, nil, rootContainer)
+
+	// Continue only if extensions feature is enabled
+	err := rootContainer.Invoke(func(alphaFeatureManager *alpha.FeatureManager) error {
+		if !alphaFeatureManager.IsEnabled(extensions.FeatureExtensions) {
+			return fmt.Errorf("extensions feature is not enabled")
+		}
+		return nil
+	})
+	if err != nil {
+		// Error here means extensions are not enabled or failed to resolve the feature manager
+		// In either case, we just proceed to normal execution
+		log.Println("auto-install extensions: ", err)
+		return rootCmd.ExecuteContext(ctx)
+	}
+
+	// Get the original args passed to the command (excluding the program name)
+	originalArgs := os.Args[1:]
+
+	// Extract flags that take values from the root command
+	flagsWithValues := extractFlagsWithValues(rootCmd)
+
+	// Find the first non-flag argument (the actual command) and check for unknown flags
+	unknownCommand, unknownFlags := findFirstNonFlagArg(originalArgs, flagsWithValues)
+
+	// If unknown flags were found before a command, return an error with helpful guidance
+	if len(unknownFlags) > 0 && unknownCommand != "" {
+		var console input.Console
+		if err := rootContainer.Resolve(&console); err != nil {
+			log.Panic("failed to resolve console for unknown flags error:", err)
+		}
+
+		flagsList := strings.Join(unknownFlags, ", ")
+		errorMsg := fmt.Sprintf(
+			"Unknown flags detected before command '%s': %s\n\n"+
+				"If you're trying to run an extension command, the extension name must come BEFORE any flags.\n"+
+				"This is because extension-specific flags are not known until the extension is installed.\n\n"+
+				"Correct usage:\n"+
+				"  azd %s %s    # Extension name first, then flags\n"+
+				"  azd %s --help          # Get help for the extension\n\n"+
+				"If this is not an extension command, please check the flag names for typos.",
```

Comment:
Is this too much of a behavioural change? If the user runs `azd --debug up`, they'll run into this error message where as before it would've worked normally.